tests/test_models.py

```diff
@@ -100,6 +100,22 @@ def test_init_with_array2():
             dm.data  # noqa: B018
 
 
+def test_init_invalid_shape():
+    """Requested some number of dimensions unequal to ndim"""
```

Comment:
You may want just mention in the docstring here what the expected dimension of `BasicModel.data` should be.